### ISSUE TITLE
Issues in GraphQL complexity analysis

### ISSUE THREAD

**Issue by RakhithaRR** [wso2/product-apim#9684]
### Description:
- Complexity analysis in graphql does not support query arguments that are not integers. For example, if we use a ENUM value as the argument of the query it will return the following error.
```
GraphQLQueryAnalysisHandler java.lang.ClassCastException: graphql.language.EnumValue cannot be cast to graphql.language.IntValue
```

- Complexity analysis only supports queries with arguments that involve limits. For example, a simple query with `id: 1000` for the argument will return the complexity limit exceeded error when rate limiting is enabled because this `id` value is considered as a limit.

eg: 
```
query{
  human(id:1000){
    id
    name
  }
}
```

### Affected Product Version:
3.2.0, 4.0.0

#### Suggested Labels:
Bug

**Comment by rmsamitha** [comment]
A new issue  https://github.com/wso2/product-apim/issues/10945 is created to track further improvement so that GraphQL API creators can define preferred keywords (for aguments) to consider as multiplying factor values, via publisher Portal

**Comment by HiranyaKavishani** [comment]
Reopening to back-port the fix to APIM 3.2.0

**Comment by dushansilva** [comment]
Closing after back-porting
